atlp-rwanda/cabal-bn-be
# #181339526 account verification

### What does this PR do?

Send verification email upon registration.

### Description of Task to be completed?
 
Send an email to a user upon successful registration. 
Users should be able to receive a verification link via email upon successful signup.
Allow only verified users to log into the system.

### How should this be manually tested?

- Clone this [repo](https://github.com/atlp-rwanda/cabal-bn-be.git)

- Checkout to this branch `feature/#181339526-account-verification`

- Type **`yarn install`** to install dependencies, and **`yarn run dbmigrate:all`** to reinitialize the database.

- Use postman and create a new user. Make sure that the email provided for the user is valid and you have access to it.

   POST

                 http://localhost:3000/api/v1/users/register
         
- Log into the email provided during registration, you should be seeing an email from Cabal Barefoot. 

- Click on a link that shows **verify**, you will be redirected to a page that shows that your account has been verified.  You can also verify your account on postman using the token sent in the link on the email.

   GET

                 http://localhost:3000/api/v1/users/verify-email/[token]


- Go back to the postman to log in. Be aware that you will only be allowed to log in after having your email verified.
                
   POST

                 http://localhost:3000/api/v1/users/login

- Documentation: [Swagger-user-verify](http://localhost:3000/api-docs/#/Users)

         


### What are the relevant pivotal tracker stories?

[#181339526](https://www.pivotaltracker.com/n/projects/2555280)

### Screenshots (if appropriate)
